searchcore/src/vespa/searchcore/proton/attribute/attribute_writer.cpp

```diff
@@ -521,20 +538,29 @@ AttributeWriter::update(SerialNum serialNum, const DocumentUpdate &upd, Document
             LOG(spam, "Failed to find attribute vector %s", fupd.getField().getName().c_str());
             continue;
         }
-        AttributeVector &attr = *attrp;
         // TODO: Check if we must use > due to multiple entries for same
         // document and attribute.
-        if (attr.getStatus().getLastSyncToken() >= serialNum)
+        if (attrp->getStatus().getLastSyncToken() >= serialNum)
             continue;
-
-        LOG(debug, "About to apply update for docId %u in attribute vector '%s'.", lid, attr.getName().c_str());
-
-        // NOTE: The lifetime of the field update will be ensured by keeping the document update alive
-        // in a operation done context object.
-        _attributeFieldWriter.execute(attr.getName(),
-                [serialNum, &fupd, lid, immediateCommit, &attr, onWriteDone]()
-                { applyUpdateToAttribute(serialNum, fupd, lid, immediateCommit, attr, onWriteDone); });
+        args[_attributeFieldWriter.getExecutorId(attrp->getName())].emplace_back(attrp, &fupd);
+        LOG(debug, "About to apply update for docId %u in attribute vector '%s'.", lid, attrp->getName().c_str());
+    }
+    // NOTE: The lifetime of the field update will be ensured by keeping the document update alive
+    // in a operation done context object.
+    for (uint32_t id(0); id < args.size(); id++) {
+        _attributeFieldWriter.execute(id, [batch = std::move(args[id])]() {
```

Comment:
Skip if task is empty.

Reply:
Also, note that execute methods calls getExecutorId() itself, you'll need to call executeLambda() instead when
providing executor id instead of component id.